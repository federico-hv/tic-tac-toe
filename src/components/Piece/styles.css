
.piece {
	width: 5em;
	height: 5em;
}

.border-1 {
	border: 3px solid white;
}

.border-2 {
	border: 3px solid black;
}

.piece-x {
	border: 3px solid black;
}

.square-0-0 {
	border-top: none;
	border-left: none;
}

.square-0-1 {
	border-top: none;
}

.square-0-2 {
	border-top: none;
	border-right: none;
}

.square-1-0 {
	border-left: none;	
}

.square-1-1 {

}

.square-1-2 {
	border-right: none;	
}

.square-2-0 {
	border-left: none;
	border-bottom: none;
}

.square-2-1 {
	border-bottom: none;
}

.square-2-2 {
	border-right: none;
	border-bottom: none;
}

.symbol-X {
	display: flex;
	justify-content: center;
	align-items: center;
	font-size: 1em;
	color: #16ECC5;
}

.symbol-O {
	display: flex;
	justify-content: center;
	align-items: center;
	font-size: 1em;
	color: #EB7272;
}


