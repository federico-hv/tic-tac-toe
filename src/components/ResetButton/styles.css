
.reset-button {
	background: #EB7272;
	color: white;
	height: 50px;
	width: 200px;
	font-family: Comic Sans MS3;
	font-size: 1.1em;
	border: 1px solid rgba(0,0,0,.2);
	margin-top: 50px;
}