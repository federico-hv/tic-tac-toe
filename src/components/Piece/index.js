import React from 'react';
import './styles.css';

const Piece = ({symbol, onSelect, squareIndex, player}) => (
	<div className={`piece square-${squareIndex} border-${player} symbol-${symbol}`} onClick={symbol === '' ? ()=>onSelect(symbol) : null}>
		{symbol}
	</div>
);


export default Piece;