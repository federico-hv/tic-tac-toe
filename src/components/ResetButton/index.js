import React from 'react';
import './styles.css';

const ResetButton = ({onAction}) => (
	<button className="reset-button" onClick={onAction}>Restart Game</button>
);

export default ResetButton;