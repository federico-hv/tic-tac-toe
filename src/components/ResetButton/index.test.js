import React from 'react';
import { mount, shallow } from 'enzyme';
import Piece from './index';
import toJson from 'enzyme-to-json';
import ResetButton from './index';


describe('<ResetButton>', ()=> {

	it('Should render a button with class reset-button and onAction prop', ()=>{
		const onAction = jest.fn();
		const button = shallow(<ResetButton onAction={onAction} />);

		expect(button.find('.reset-button').length).toEqual(1);
		button.simulate('click');
		expect(onAction.mock.calls.length).toBe(1);
	})

	it('Should Match snaptshot', ()=>{
		const button = shallow(<ResetButton />);

		expect(toJson(button)).toMatchSnapshot();
		button.unmount();
	})
});