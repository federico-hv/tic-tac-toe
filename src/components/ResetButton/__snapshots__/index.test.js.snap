// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetButton> Should Match snaptshot 1`] = `
<button
  className="reset-button"
>
  Restart Game
</button>
`;
